src/sources/soundsourceffmpeg.cpp

```diff
@@ -79,15 +79,15 @@ inline int64_t getStreamStartTime(const AVStream& avStream) {
     auto start_time = avStream.start_time;
     if (start_time == AV_NOPTS_VALUE) {
         // This case is not unlikely, e.g. happens when decoding WAV files.
-#if ENABLE_TRACING
-        kLogger.trace()
+#if ENABLE_TRACE_LOG
```

Comment:
Can't we get rid of this?

I don't think this makes a notable performance improvement:
```
Note: Arguments aren't processed if the debug output for that category is not enabled, so don't rely on any side effects.
```

Reply:
Qt does not provided a detailed trace level, only debug. Until the default log level has been switched from *debug* to *info* we cannot avoid this conditional compilation to avoid log spam.

And even then log messages in real-time threads need to be disabled at compile time by using conditional compilation. That's the reason why I also picked an example from the sound device code.

Reply:
Why do we need something more detailed than debug? IMHO it makes more sense to switch it on for the logging category you're interested in and use INFO for everything else by default.

Why do they need to be disabled in RT code? If arguments aren't processed if debug logging is disabled (as the documentation states), shouldn't this just cause one additional conditional branch per log statement? That should be OK even in performance-critical code, right?

Reply:
String operations are too slow for realtime code.

Reply:
The logs we are talking about in the real-time code are only used for tracing execution paths in the callback loop during development. These are invoked frequently and should **not** be compiled into any executable version, even if they would be real-time safe.

The Qt logging facilities will have some performance impact at runtime. And they have been designed for general purpose desktop UI applications, not for performance critical code.

Reply:
I will add a global define to enable verbose debug logs like in `SoundSourceFFmpeg`. These could be enabled permanently once we have switched the default log level from *Debug* to *Info*.

The local defines for trace logs in `SoundDevicePortAudio` are still needed as explained. I will try to add a reference to this example as blueprint.